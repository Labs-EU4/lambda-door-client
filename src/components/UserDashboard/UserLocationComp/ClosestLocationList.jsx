/* eslint-disable no-shadow */
import React from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import { Spin } from 'antd';
import CompanySummaryCard from '../../CompanySummaryCard';

export const ClosestLocationList = ({
  isFetching,
  closestCompanies: { closestCompanies },
}) => {
  return (
    <StyledContainer>
      {!isFetching ? (
        <>
          {closestCompanies.length !== 0 ? (
            closestCompanies.map(closest => (
              // eslint-disable-next-line react/no-array-index-key
              <CompanySummaryCard
                key={`${closest.id}`}
                text={closest.description}
                name={closest.name}
                id={closest.id}
                website={closest.website}
                rating={closest.rating}
              />
            ))
          ) : (
            <div className="empty-state">
              <p>No data to display</p>
            </div>
          )}
        </>
      ) : (
        <div className="empty-state">
          <Spin />
        </div>
      )}
    </StyledContainer>
  );
};

export default connect(state => state, {})(ClosestLocationList);

const StyledContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-gap: 1.5rem;
  min-height: 100%;
  position: relative;
  min-height: 150px;

  @media (max-width: 1280px) {
    grid-template-columns: repeat(2, 1fr);
  }

  @media (max-width: 768px) {
    grid-template-columns: repeat(1, 1fr);
  }

  .cards {
    position: relative;
    height: 150px;
    max-width: 350px;

    a.nav-link {
      position: absolute;
      left: 0;
      right: 0;
      top: 0;
      bottom: 0;
      padding: 1rem;
      color: rgba(0, 0, 0, 0.65);
      border: 1px solid #e8e8e8;
      border-radius: 2px;
      display: flex;
      flex-direction: column;
      justify-content: space-between;

      &:hover {
        border: 1px solid #bb1333;
      }
    }

    h3 {
      font-weight: 600;
      font-size: 1rem;
    }

    .ant-rate,
    .ant-rate-star-first,
    .ant-rate-star-second {
      height: 20px;
      width: fit-content;
      display: inline-flex;
      align-items: center;
      justify-content: center;
    }

    .ant-rate-star-first i,
    .ant-rate-star-second i {
      font-size: 14px;
    }

    .ant-rate-star:not(:last-child) {
      margin-right: 3px;
    }
  }
`;
