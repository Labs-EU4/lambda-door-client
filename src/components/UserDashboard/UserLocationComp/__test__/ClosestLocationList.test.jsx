import React from 'react';
import * as rtl from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter as Router } from 'react-router-dom';
import { ClosestLocationList } from '../ClosestLocationList';

const getClosestCompanies = jest.fn();
const isFetching = false;
const closestCompanies = {
  closestCompanies: [
    {
      name: 'Accenture',
      website: 'www.accenture.com.',
      location: 'Atlanta, GA',
      longitude: -85.0,
      latitude: 33.7537,
      type: 'Business',
      logo: '',
      description: '',
    },
  ],
};
let wrapper;
beforeEach(() => {
  rtl.cleanup();
  wrapper = rtl.render(
    <Router>
      <ClosestLocationList
        getClosestCompanies={getClosestCompanies}
        isFetching={isFetching}
        closestCompanies={closestCompanies}
      />
    </Router>
  );
});

describe('ClosestLocationList', () => {
  wrapper = rtl.render(
    <Router>
      <ClosestLocationList
        getClosestCompanies={getClosestCompanies}
        isFetching={isFetching}
        closestCompanies={closestCompanies}
      />
    </Router>
  );
  it('renders correctly', () => {
    expect(wrapper.baseElement).toMatchSnapshot();
  });
});
