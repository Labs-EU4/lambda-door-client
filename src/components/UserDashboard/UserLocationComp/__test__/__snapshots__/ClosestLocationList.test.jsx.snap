// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClosestLocationList renders correctly 1`] = `
<body>
  <div>
    <div
      class="sc-bxivhb iVKdRw"
    >
      <div
        class="cards"
      >
        <a
          class="nav-link"
          href="/company-page/undefined"
        >
          <div>
            <h3>
              Accenture
            </h3>
            <p
              class="sc-bwzfXH hKkmqJ"
            />
          </div>
        </a>
        <div
          class="sc-htpNat ckDsGN"
        >
          <div />
          <div>
            <a
              class="sc-bdVaJa jHdICt"
              href="www.accenture.com."
              rel="noopener noreferrer"
              target="_blank"
            >
              www.accenture.com.
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
